### PR TITLE
Change back from static field calls

### PR DESCRIPTION
**Pull request by davisshaver**
Follows up https://github.com/alleyinteractive/wp-seo/pull/63

Supported by https://github.com/alleyinteractive/wp-seo/pull/72

### CODE REVIEW

File: php/class-wp-seo-social-wp-seo-filters.php [hunk 1]
```diff
@@ -53,6 +55,84 @@ public function filter_wp_seo_options_page_menu_title() {
...
+	 */
+	public function filter_wp_seo_sanitize_as_integer_field() {
+		$sanitize_as_integer = array(
+			'home_og_image',
+		);
+		foreach ( WP_SEO_Settings()->single_post_types as $type ) {
+			$sanitize_as_integer[] = "single_{$type->name}_og_image";
+		}
+		// Post type, taxonomy, and other archives.
+		foreach ( array_merge( WP_SEO_Settings()->archived_post_types, WP_SEO_Settings()->taxonomies ) as $type ) {
```
Comment (kjbenk) [inline php/class-wp-seo-social-wp-seo-filters.php:81]: Can you be 100% certain that both of those functions will return arrays?  If not you might need to type cast to use `array_merge` without a notice.

File: php/class-wp-seo-social-wp-seo-filters.php [hunk 2]
```diff
@@ -53,6 +55,84 @@ public function filter_wp_seo_options_page_menu_title() {
...
+			'home_og_description',
+			'home_og_type',
+		);
+		foreach ( WP_SEO_Settings()->single_post_types as $type ) {
+			$sanitize_as_text_field[] = "single_{$type->name}_og_title";
+			$sanitize_as_text_field[] = "single_{$type->name}_og_description";
+			$sanitize_as_text_field[] = "single_{$type->name}_og_type";
+		}
+		// Post type, taxonomy, and other archives.
+		foreach ( array_merge( WP_SEO_Settings()->archived_post_types, WP_SEO_Settings()->taxonomies ) as $type ) {
```
Comment (kjbenk) [inline php/class-wp-seo-social-wp-seo-filters.php:112]: Same as above.

**Review by kjbenk — CHANGES_REQUESTED** [review]
Minor comment, otherwise it is good.

File: php/class-wp-seo-social-wp-seo-filters.php (hunk 1 above)
Comment (davisshaver) [reply php/class-wp-seo-social-wp-seo-filters.php:81]: Yes will return array always

**Pull request merged**